### PR TITLE
force garbo to only use Skill and Item inputs to macros

### PR DESCRIPTION
**Pull request by horrible-little-slime**

### CODE REVIEW

**Comment by horrible-little-slime** [comment]
This will likely be obsoleted by the next libram update.

File: src/combat.ts [hunk 1]
```diff
@@ -134,12 +134,49 @@ function shouldRedigitize() {
...
   submit(): string {
     print(this.components.join("\n"));
     return super.submit();
   }
 
+  skill(...skills: Skill[]): this {
+    return super.skill(...skills);
+  }
+
+  static skill<T extends LibramMacro>(this: Constructor<T>, ...skills: Skill[]): T {
```
Comment (phulin) [inline src/combat.ts:149]: should be `T extends Macro`, no?

**Comment by horrible-little-slime** [comment]
I got red squigglies when I did that, but didn't get that for StrictMacro
in Libram. Maybe relates to reusing the name?

File: src/combat.ts (hunk 1 above)
Comment (horrible-little-slime) [reply src/combat.ts:149]: It seems like, for whatever reason, VSCode hates that (red squigglies), but I can do it in my libram PR. I'd be willing to guess the issue is recycling the name 'Macro.' I'll close this PR and wait until we get StrictMacro in libram.

**Pull request closed without merging**